aptos-move/framework/supra-framework/sources/pbo_delegation_pool.move

```diff
@@ -4726,4 +5601,468 @@ module supra_framework::pbo_delegation_pool {
+    #[expected_failure(abort_code = 327716, location = Self)]
+    /// if admin is multi signer calling `replace_delegator` if it's not the same signer which was initialized, it should fail
+    public entry fun test_repalce_delegration_with_different_multisig_failure(
+        supra_framework: &signer,
+        validator: &signer
+    ) acquires DelegationPoolOwnership, DelegationPool, GovernanceRecords, BeneficiaryForOperator, NextCommissionPercentage {
+        initialize_for_test(supra_framework);
+        account::create_account_for_test(signer::address_of(validator));
+        let delegator_address = vector[@0x010, @0x020];
+        let principle_stake = vector[100 * ONE_APT, 200 * ONE_APT];
+        let coin = stake::mint_coins(300 * ONE_APT);
+        let principle_lockup_time = 0;
+        let multisig = generate_multisig_account(validator, vector[@0x12134], 2);
+
+        initialize_test_validator(validator, 0, true, true, 0,
+            delegator_address,
+            principle_stake,
+            coin,
+            option::some(multisig),
+            vector[2, 2, 3],
+            10,
+            principle_lockup_time,
+            12
+        );
+
+        let new_multisig = generate_multisig_account(supra_framework, vector[@0x12234], 2);
+        let multisig_signer = account::create_signer_for_test(new_multisig);
+
+        let validator_address = signer::address_of(validator);
+        let pool_address = get_owned_pool_address(validator_address);
+        replace_delegator(&multisig_signer, pool_address, @0x010, @0x0101);
+    }
+
+    #[test(supra_framework = @supra_framework, validator = @0x123)]
+    /// if admin is authorized multi signer, `replace_delegator` should succeed
+    public entry fun test_repalce_delegration_multisig_success(
+        supra_framework: &signer,
+        validator: &signer
+    ) acquires DelegationPoolOwnership, DelegationPool, GovernanceRecords, BeneficiaryForOperator, NextCommissionPercentage {
+        initialize_for_test(supra_framework);
+        account::create_account_for_test(signer::address_of(validator));
+        let delegator_address = vector[@0x010, @0x020];
+        let principle_stake = vector[100 * ONE_APT, 200 * ONE_APT];
+        let coin = stake::mint_coins(300 * ONE_APT);
+        let principle_lockup_time = 0;
+        let multisig = generate_multisig_account(validator, vector[@0x12134], 2);
+
+        initialize_test_validator(validator, 0, true, true, 0,
+            delegator_address,
+            principle_stake,
+            coin,
+            option::some(multisig),
+            vector[2, 2, 3],
+            10,
+            principle_lockup_time,
+            12
+        );
+
+        let multisig_signer = account::create_signer_for_test(multisig);
+        let validator_address = signer::address_of(validator);
+        let pool_address = get_owned_pool_address(validator_address);
+        replace_delegator(&multisig_signer, pool_address, @0x010, @0x0101);
```

Comment:
Check after replacement that `new_delegator` has the same `active_shares` `inactive_shares`, `pending_withdrawal`, `principle_stake` as `old_delegator` before replacement here. Without which we are not really asserting/validating anything here. @nizam-supraoracles